MyFirstApplication/Services/Services.cs

```diff
@@ -11,32 +11,32 @@ namespace MyFirstApplication.Services
     public class Services : IServices
     {
         private readonly HttpClient _httpClient;
-        private readonly IConfiguration _configuration;
+        private readonly IOptions<AppSettings> _appSettings;
 
-        public Services(IHttpClientFactory httpClientFactory, IConfiguration configuration)
+        public Services(IHttpClientFactory httpClientFactory, IOptions<AppSettings> appSettings)
         {
             _httpClient = httpClientFactory.CreateClient();
-            _configuration = configuration;
+            _appSettings = appSettings;
         }
-        public async Task<IList<Show>> GetListOfShows()
+        public async Task<IList<TvShow>> GetListOfTvShows()
         {
-
-
-            var baseUri = _configuration["URL:BaseURI"];
-            HttpResponseMessage response = await _httpClient.GetAsync(baseUri);
+            
+            var baseURI = _appSettings.Value.BaseURI;
+            //var baseUri = _configuration["URL:BaseURI"];
+            var response = await _httpClient.GetAsync(baseURI);
 
             if (response.IsSuccessStatusCode)
             {
                 var res = await response.Content.ReadAsStringAsync();
-                var showIndex = JsonConvert.DeserializeObject<IList<Show>>(res);
+                var showIndex = JsonConvert.DeserializeObject<IList<TvShow>>(res);
 
                 if (showIndex != null)
                 {
                     return showIndex;
                 }
             }
 
-            return new List<Show>();
+            return new List<TvShow>();
```

Comment:
Once the above changes are made, you can then inject the `TvShowsHttpClient` in `TvShowsService` like below. You can also use a primary constructor, instead of defining a separate constructor. This would allow you to just use the injected variable directly instead of defining `private readonly` variables.

You TvShowService will then be reduced to just the HttpClient call. As all the deserialization logic now moved to HttpClient, conforming to S in the SOLID i.e. Single Responsiblity Principle.

```
public class TvShowService(TvShowsHttpClient client) : ITvShowService
{
    public async Task<IList<Show>> GetListOfShows()
    {
          return await client.GetTvShows();
    }
}

Reply:
There is even a better way using the `HttpClient` without worrying about the deserialization. But let's leave it for now. 

However, you need to find a way of identifying if the response was successful or not in the TvShowService, but still leaving the deserialization part in the TvShowsHttpClient.

The reason is that HttpClient call to thirdparty api could fail, and so it would return error information. That error information could potentially be logged, so that you can identify later that there was a problem, but for user you need to display an friendly error page, telling user that there was some problem, so try again.